Components/RTShaderSystem/src/OgreShaderGLSLProgramWriter.cpp

```diff
@@ -60,6 +60,7 @@ void GLSLProgramWriter::initializeStringMaps()
     mGpuConstTypeMap[GCT_SAMPLERCUBE] = "samplerCube";
     mGpuConstTypeMap[GCT_SAMPLER1DSHADOW] = "sampler1DShadow";
     mGpuConstTypeMap[GCT_SAMPLER2DSHADOW] = "sampler2DShadow";
+    mGpuConstTypeMap[GCT_SAMPLER_EXTERNAL_OES] = "samplerExternalOES";
```

Comment:
now that we got this, you could add writing
```cpp
#extension GL_OES_EGL_image_external : require
```
after this block

https://github.com/OGRECave/ogre/blob/1074eaab13d3856774f08aed50064b7b62d18011/Components/RTShaderSystem/src/OgreShaderGLSLESProgramWriter.cpp#L202-L207

so the RTSS auto-generates the correct shader and remove the custom shaders again. But this change is optional.

Also you would need to add the sampler here:

https://github.com/OGRECave/ogre/blob/1074eaab13d3856774f08aed50064b7b62d18011/Components/RTShaderSystem/src/OgreShaderGLSLESProgramWriter.cpp#L343-L344

Reply:
tried it.. GLSLES drivers dont like the FFP_SampleTexture indirection. Must get rid of it first